ambuda/views/auth.py

```diff
@@ -79,51 +91,99 @@ def _is_valid_reset_token(row: db.PasswordResetToken, raw_token: str, now=None):
     return True
 
 
+# Copied from https://wtforms.readthedocs.io/en/2.3.x/validators/
+class FieldLength(object):
+    def __init__(self, min=-1, max=-1, message=None):
```

Comment:
rather than -1 for sentinel values, let's use `None`

Reply:
Seems like logically and functionally -1 is a better solution. Any particular concern? I don't mind changing, but with None we have to change this line:
`if field_len < self.min or self.max != -1 and field_len > self.max:`
or
` if self.min and field_len < self.min or self.max and field_len > self.max:`

Reply:
I like `None` because `self.x < None` will `TypeError` and standard linters will check that the `None` case is handled. At init time we can do:

```
self.min = min or 0
self.max = max or sys.maxint
```

Then our checks can be:

```
if not (self.min <= field_len <= self.max_len):
```

Reply:
@kvchitrapu note that with this approach ^ we handle `None` only at init time and we get the benit of "ints everywhere" elsewhere without having to remember to check for -1

Reply:
> I like `None` because `self.x < None` will `TypeError` (i.e. it will fail fast) and standard linters will check that the `None` case is handled. At init time we can do:
> 
> ```
> self.min = min or 0
> # EDIT: should be `sys.maxsize` in python3
> self.max = max or sys.maxsize
> ```
> 
> Then our checks can be:
> 
> ```
> if not (self.min <= field_len <= self.max):
> ```

Good. Now I see what you did here. Like it!